fetching/strat.py

```diff
@@ -61,28 +59,15 @@ def __init__(
         self.pytest_root = pytest_root
         self.decorator_appended_file_paths: list[pathlib.Path] = []
 
-        self.sys_path_ext = f"sys.path.append('{self.pytest_root}')\n"
-
     def _apply(self, path: pathlib.Path) -> None:
         with path.open() as file:
-            lines = file.readlines()
-
-        skip_line = False
-        contains_pytest_test_function = False
-        for i, line in enumerate(lines):
-            if skip_line:
-                skip_line = False
-                continue
-            if PyTestStrategy.FUNCTION_PATTERN.fullmatch(line):
-                lines.insert(i, PyTestStrategy.REGISTER)
-                skip_line = True
-                contains_pytest_test_function = True
-
-        if contains_pytest_test_function:
-            lines.insert(0, PyTestStrategy.SYS_IMPORT)
-            lines.insert(1, self.sys_path_ext)
-            lines.insert(2, PyTestStrategy.PYTYPE_IMPORTS)
-            lines.append(PyTestStrategy.ENTRYPOINT)
+            file_content = file.read()
+
+        append_decorators_transformer = AppendDecoratorsTransformer(PyTestStrategy.FUNCTION_PATTERN, self.pytest_root)
+        file_ast = ast.parse(file_content)
+        file_ast = append_decorators_transformer.visit(file_ast)
+        append_decorators_transformer.append_nodes_necessary_for_tracing(file_ast)
```

Comment:
Move this method into `PyTestStrategy`.
Visitors should only `visit`.

Reply:
It is a transformer, it changes the AST. Also I think it is more logical to define it in the transformer as nodes are appended to the AST = The AST is modified.

Reply:
Visitors only have visit methods on them. This is known as the Visitor Pattern

Reply:
It is not a visitor. It is a transformer.

Reply:
ah, this is slightly hidden in the docs. 
`ast.NodeTransformer` inherits from `ast.NodeVisitor`, meaning the transformer IS a visitor.

Reply:
Alright